Handle differing emails on parse and S3

gmartin7 opened the pull request:
* also prettify Readme

<!-- Reviewable:start -->
---
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/bloombooks/bloombulkdownloader/9)
<!-- Reviewable:end -->

Commit:
Handle differing emails on parse and S3

* also prettify Readme
* added url decode test

andrew-polk commented:
Wondering if we need any tests for encoded characters in urls. Perhaps they are already there prior to this PR.

---

Reviewed 6 of 6 files at r1.
Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[BloomBulkDownloader/BulkDownload.cs, line 149 at r1](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-KtWrPvQkQPWJO5Z96K0:-KtWrPvQkQPWJO5Z96K1:be3v04b) ([raw file](https://github.com/bloombooks/bloombulkdownloader/blob/bf6929448e9262fdb8bdd2a37564ddca60227edd/BloomBulkDownloader/BulkDownload.cs#L149)):*
> recordList

I find myself wishing this had a more descriptive variable name.

---

*[BloomBulkDownloader/BulkDownload.cs, line 154 at r1](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-KtWqaxOzekJBz56vcWK:-KtWqaxOzekJBz56vcWL:bheq9jm) ([raw file](https://github.com/bloombooks/bloombulkdownloader/blob/bf6929448e9262fdb8bdd2a37564ddca60227edd/BloomBulkDownloader/BulkDownload.cs#L154)):*
> recordList[0]

Perhaps nitpicking, but this probably deserves a local variable.

---

*[BloomBulkDownloader/BulkDownload.cs, line 162 at r1](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-KtWqS081cmk-n1MIsgw:-KtWqS081cmk-n1MIsgx:bkyuv7u) ([raw file](https://github.com/bloombooks/bloombulkdownloader/blob/bf6929448e9262fdb8bdd2a37564ddca60227edd/BloomBulkDownloader/BulkDownload.cs#L162)):*
> dummy

I wouldn't call this "dummy" as it is used below.
Same for else clause later.

---

*[BloomBulkDownloaderTests/DownloaderTests.cs, line 22 at r1](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-KtWsX3bgQDtSgEQ7G6h:-KtWsX3bgQDtSgEQ7G6i:b-wz9ecw) ([raw file](https://github.com/bloombooks/bloombulkdownloader/blob/bf6929448e9262fdb8bdd2a37564ddca60227edd/BloomBulkDownloaderTests/DownloaderTests.cs#L22)):*
> ```Smalltalk
> 		public DownloaderTests()
> 		{
> 			_oct102015 = new DateTime(2015, 10, 10);
> ```

I think these would make more sense as constants.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-:-KtWt2uSoziMZpRCgnCF:bikjgoc)*
<!-- Sent from Reviewable.io -->

gmartin7 commented:
Added a test of the DecodeBaseUrl() method that shows comma, nbsp, schwa and barred i all decode correctly from the baseUrl.

---

Review status: all files reviewed at latest revision, 4 unresolved discussions.

---

*[BloomBulkDownloader/BulkDownload.cs, line 149 at r1](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-KtWrPvQkQPWJO5Z96K0:-KtXF9_15955QYrPaaGS:b-896fix) ([raw file](https://github.com/bloombooks/bloombulkdownloader/blob/bf6929448e9262fdb8bdd2a37564ddca60227edd/BloomBulkDownloader/BulkDownload.cs#L149)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> recordList

I find myself wishing this had a more descriptive variable name.
</blockquote></details>

Done.

---

*[BloomBulkDownloader/BulkDownload.cs, line 154 at r1](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-KtWqaxOzekJBz56vcWK:-KtXI3fJALtZw5JMMKpO:b-896fix) ([raw file](https://github.com/bloombooks/bloombulkdownloader/blob/bf6929448e9262fdb8bdd2a37564ddca60227edd/BloomBulkDownloader/BulkDownload.cs#L154)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> recordList[0]

Perhaps nitpicking, but this probably deserves a local variable.
</blockquote></details>

Done.

---

*[BloomBulkDownloader/BulkDownload.cs, line 162 at r1](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-KtWqS081cmk-n1MIsgw:-KtXIMdiv13DRSj1m1KN:b-eof7s8) ([raw file](https://github.com/bloombooks/bloombulkdownloader/blob/bf6929448e9262fdb8bdd2a37564ddca60227edd/BloomBulkDownloader/BulkDownload.cs#L162)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

> dummy

I wouldn't call this "dummy" as it is used below.
Same for else clause later.
</blockquote></details>

chgd to existingBookByUrl.
I also extracted the common code in the if-else out to UpdateDictWithMostRecentMatch().

---

*[BloomBulkDownloaderTests/DownloaderTests.cs, line 22 at r1](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-KtWsX3bgQDtSgEQ7G6h:-KtXUWwvxOEOoLW932HA:bbgsbad) ([raw file](https://github.com/bloombooks/bloombulkdownloader/blob/bf6929448e9262fdb8bdd2a37564ddca60227edd/BloomBulkDownloaderTests/DownloaderTests.cs#L22)):*
<details><summary><i>Previously, andrew-polk wrote…</i></summary><blockquote>

I think these would make more sense as constants.
</blockquote></details>

A DateTime can't be a const in C#. I can put in a string const like: "2015-10-10T09:00:00" and then parse that into the existing variable in the DownloaderTests ctor, but I'm not sure that gains anything.

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-:-KtXnVToczuJfAvCpTw4:bw4ikwu)*
<!-- Sent from Reviewable.io -->

andrew-polk commented:
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

---

Reviewed 2 of 2 files at r2.
Review status: all files reviewed at latest revision, all discussions resolved.

---



*Comments from [Reviewable](https://reviewable.io:443/reviews/bloombooks/bloombulkdownloader/9#-:-KtXqM8MRPkuJOi6RqgU:bnfp4nl)*
<!-- Sent from Reviewable.io -->
